repeat_expansion_variants/README.md

```diff
@@ -0,0 +1,122 @@
+
+### Step 2. Parse variant identifiers
+The **Name** column contains a variant description which can be in three possible formats:
+1. HGVS-like variant description in coding or genomic coordinates, e. g. `NM_000044.4(AR):c.172_174CAG(10_36) (p.Gln69_Gln80del)`.
+2. HGVS-like variant description in protein coordinates, e. g. `NP_003915.2:p.Ala260(5_9)`.
+3. Human-readable, however standardised, description, e. g. `ATN1, (CAG)n REPEAT EXPANSION`.
+
+There are also three properties which we would like to extract:
+1. Lengh of the repeat unit, determined directly from the sequence. In examples 1 and 3 it would be len(CAG) = 3.
+2. Length of the repeat unit, determined indirectly from the coordinate span of the variant. In example 1, it would be 174 – 172 + 1 = 3.
+3. Transcript in which the variant resides. In example 1, it would be NM_000044.
+
+Here is the table describing which properties can be extracted from which identifier types:
+
+| Identifier type           | Repeat unit length | Coordinate span | Transcript ID |
+| ------------------------- | ------------------ | --------------- | ------------- |
+| HGVS-like, coding/genomic | sometimes*         | sometimes*      | yes           |
+| HGVS-like, protein        | –                  | –               | –             |
+| Free text                 | yes                | –               | –             |
+
+_\* At least one of (repeat unit length, coordinate span) is always present in this case_
+
+At this step, we extract all properties which we can from the variant identifiers and save them to separate dataframe columns.
+
+### Step 3. Annotate Ensembl gene information
+There are three columns which can be used to deduce gene information: HGNC_ID & GeneSymbol (from ClinVar data), and TranscriptID (parsed from the variant identifier). Because neither of them is present for **all** records, we successively try to annotate Ensembl gene ID based on each of those columns in that order. For example, if Ensembl gene ID for a particular record was determined based on HGNC_ID, then GeneSymbol and TranscriptID will not be tried for that column. Ensembl gene information is being annotated using their [BioMart service](https://www.ensembl.org/biomart/).
+
+In some cases, a mapping from HGNC ID produces multiple Ensembl hits. For example, HGNC:10560 is being resolved to ENSG00000285258 and ENSG00000163635. In this case, a record is further split into two, each mapping into their corresponding Ensembl gene IDs.
+
+### Step 4. Determine repeat type and record completeness
+There are two repeat types detected by this pipeline:
+* `trinucleotide_repeat_expansion`, corresponding to SO term http://sequenceontology.org/browser/current_release/term/SO:0002165. This is chosen when the repeat length appears to be either 3, **or a multiple or 3.** For example:
```

Comment:
> or a **multiple** of 3
Is this additional OR criteria defined in relevant literature? I can't see it in the said link but if there is some other documentation which cites this I would suggest linking to that too....

Reply:
This is actually something which I want to discuss with Open Targets directors. When I will be sending an update about this pipeline, I will include this specific question about whether we should include strictly trinucleotide repeats, or those which are a multiple of 3.